lib/rate-limit.js

```diff
@@ -81,5 +81,9 @@ function startDrainer() {
 }
 
 function interval() {
-   return (resetAt * 1000 - Date.now()) / (remaining + 1);;
+   var requestsRemaining = Math.max(1, remaining);
+   // Always underconsume, and leave a minute's worth of requests remaining
+   // just in case our clock drifts from githubs
+   var timeTillReset = Math.max(1, (60 + resetAt) * 1000 - Date.now());
```

Comment:
Gotcha, I was confused by the global variables. Is [`resolve`](https://github.com/iFixit/pulldasher/pull/35/files#diff-80d900d5fc6a583850c7b52862522c34R62) supposed to be a global variable? I don't see where that's defined.